pagination-practice/src/components/PaginatedTable.jsx

```diff
@@ -0,0 +1,182 @@
+import { useEffect, useState } from "react";
+import axios from "axios";
+export default function PaginatedTable() {
+  const [data, setData] = useState([]);
+  const [currentData, setCurrentData] = useState([]);
+  const [currentPageIndex, setCurrentPageIndex] = useState(0);
+  const [pageSize, setPageSize] = useState(10);
+
+  const pageCount = Math.ceil(data.length / pageSize);
+
+  const indexOfFirstItem = pageSize * currentPageIndex;
+  const indexOfLastItem = Math.min(
+    indexOfFirstItem + pageSize,
+    data.length - 1
+  );
+
+  function gotoPage(pageIndex) {
+    if (pageIndex >= 0 && pageIndex < pageCount) {
+      setCurrentPageIndex(pageIndex);
+    }
+  }
+
+  function canGotoPreviousPage() {
+    if (currentPageIndex > 0) {
+      return true;
+    }
+    return false;
+  }
+
+  function canGotoNextPage() {
+    console.log("PageCount", pageCount);
+    if (currentPageIndex < pageCount - 1) {
+      return true;
+    }
+    return false;
+  }
+
+  function previousPage() {
+    if (canGotoPreviousPage()) {
+      setCurrentPageIndex(currentPageIndex - 1);
+    }
+  }
+
+  function nextPage() {
+    if (canGotoNextPage()) {
+      setCurrentPageIndex(currentPageIndex + 1);
+    }
+  }
+
+  useEffect(() => {
+    axios.get("https://api.spacexdata.com/v5/launches").then((res) => {
+      setData(res.data);
+    });
+  }, []);
+
+  useEffect(() => {
+    setCurrentData(
+      data.filter(
+        (item, index) => index >= indexOfFirstItem && index <= indexOfLastItem
+      )
+    );
+  }, [data, indexOfFirstItem, indexOfLastItem]);
+
+  const headers = ["ID", "Name", "Flight Number", "Date", "Success status"];
+
+  function renderTableHeader() {
+    return (
+      <thead className="text-lg text-gray-900 bg-gray-200">
+        <tr>
+          {headers.map((header) => (
+            <th scope="col" className="px-6 py-2" key={header}>
+              {header}
+            </th>
+          ))}
+        </tr>
+      </thead>
+    );
+  }
+
+  function renderTableData() {
+    //console.log(currentData);
+    return (
+      <tbody>
+        {currentData.map((item) => (
+          <tr className="bg-white border-b" key={item.id}>
```

Comment:
I think we can optimize this by mapping with key value pair. Then we don't need to write <td > several times.

Reply:
updated